lib/cuda/cudaDeviceInterface.cpp

```diff
@@ -24,6 +42,22 @@ cudaDeviceInterface::~cudaDeviceInterface() {
     cleanup_memory();
 }
 
+void cudaDeviceInterface::set_device_clocks(const string& unique_name, int32_t gpu_id_) {
+
+    // Set default clocks to zero
+    uint32_t gpu_mem_clock = config.get_default<uint32_t>(unique_name, "gpu_core_clock", 1000);
+    uint32_t gpu_core_clock = config.get_default<uint32_t>(unique_name, "gpu_core_clock", 1000);
```

Comment:
I would set the default to `0`, and then check below these lines if the value is zero or not.   If one or both is, then don't set the clocks, otherwise apply the clocks.   That way if someone doesn't provide a set of clocks in the config, it won't try to set anything and just use the GPU's default.

Reply:
Set the clock values to `0` and applied the conditions for keeping the values zero if the clock values from the config files are not matching the allowed GPU clocks.